packages/components/src/heading/stories/index.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
+
+/**
+ * Internal dependencies
+ */
+import { Heading } from '..';
+
+const meta: ComponentMeta< typeof Heading > = {
+	component: Heading,
+	title: 'Components (Experimental)/Heading',
+	argTypes: {
+		adjustLineHeightForInnerControls: { control: { type: 'text' } },
```

Comment:
`adjustLineHeightForInnerControls` can also be a `boolean` — should we add `true` and `false` to the list of possible values, maybe in a dropdown-like control?

Reply:
Mmm yeah, this one’s a bit overcomplicated too. Based on the [code](https://github.com/WordPress/gutenberg/blob/storybook/prop-type-extraction/packages/components/src/text/get-line-height.ts), `true` gives the same result as `'medium'`, and `false` gives the same result as `undefined`. Before we graduate `<Text>` from experimental status, I'd prefer removing the boolean types from this prop API. (Kind of similar to your `UnitControl` cleanup!)

I'll note this in the TS refactor tracking issue.

Reply:
Just to close the circle, this was finally done in #54953 !